src/go/rpk/pkg/cli/cloud/login.go

```diff
@@ -60,36 +67,167 @@ token and client ID is always synced.
 		Run: func(cmd *cobra.Command, _ []string) {
 			cfg, err := p.Load(fs)
 			out.MaybeDie(err, "unable to load config: %v", err)
+			y := cfg.VirtualRpkYaml()
+			auth := y.Auth(y.CurrentCloudAuth)
+			cc := auth.HasClientCredentials()
 
 			_, err = oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg.DevOverrides()))
-			if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) {
-				out.Die(`
-Unable to login into Redpanda Cloud: %v.
-
-You may need to clear your client ID and secret with 'rpk cloud logout --clear-credentials',
-and then re-specify the client credentials next time you log in.`, err)
+			if err != nil {
+				if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) && cc {
+					fmt.Println(`You may need to clear your client ID and secret with 'rpk cloud logout --clear-credentials',
```

Comment:
the fmt change was because I meant to split this message into two lines, with this second line in a conditional -- so I figured `out.Die` once looked better, after the fmts